crates/red_knot_python_semantic/resources/mdtest/snapshots/overloads.md_-_Overloads_-_Invalid_-_At_least_two_overloads.snap

```diff
@@ -0,0 +1,59 @@
+---
+source: crates/red_knot_test/src/lib.rs
+expression: snapshot
+---
+---
+mdtest name: overloads.md - Overloads - Invalid - At least two overloads
+mdtest path: crates/red_knot_python_semantic/resources/mdtest/overloads.md
+---
+
+# Python source files
+
+## mdtest_snippet.py
+
+```
+ 1 | from typing import overload
+ 2 | 
+ 3 | @overload
+ 4 | def func1(x: int) -> int: ...
+ 5 | 
+ 6 | # error: [invalid-overload]
+ 7 | def func1(x: int | str) -> int | str:
+ 8 |     return x
+ 9 | 
+10 | @overload
+11 | # error: [invalid-overload]
+12 | def func2(x: int) -> int: ...
+```
+
+# Diagnostics
+
+```
+error: lint:invalid-overload: Function `func1` requires at least two overloads
```

Comment:
nit: This reads slightly off to me, because the function doesn't _require_ at least two overloads, it could have zero and that would also be fine. I think one way to address this would be to say "Overloaded function `func1`..." instead of just "Function `func1`..." -- if it is specified that it is an overloaded function, then it is true that it requires at least two overloads.

Reply:
Yeah, that makes sense. Thanks!